README.Rmd

```diff
@@ -743,6 +746,25 @@ Both [Tidymodels](https://tune.tidymodels.org/articles/extras/optimizations.html
 * The number of threads is set via the [num_threads](https://lightgbm.readthedocs.io/en/latest/Parameters.html#num_threads) parameter, which is passed to the model using the `set_args()` function from `parsnip`. By default, `num_threads` is equal to the full number of physical cores available. More (or faster) cores will decrease total training time.
 * This repository uses the CPU version of LightGBM included with the [LightGBM R package](https://lightgbm.readthedocs.io/en/latest/R/index.html). If you'd like to use the GPU version you'll need to [build it yourself](https://lightgbm.readthedocs.io/en/latest/R/index.html#installing-a-gpu-enabled-build) or wait for the [upcoming CUDA release](https://github.com/microsoft/LightGBM/issues/5153).
 
+## Updating R dependencies
+
+There are two lockfiles that we use with renv to manage R dependencies:
+
+1. **`renv.lock`** is the canonical list of dependencies that are used by the **core model pipeline**. Any dependencies that are required to run the model itself should be defined in this lockfile.
+2. **`renv/profiles/reporting/renv.lock`** is the canonical list of dependencies that are used to **generate a model performance report** in the `finalize` step of the pipeline. Any dependencies that are required to generate that report or others like it should be defined in this lockfile.
+
+Our goal in maintaining multiple lockfiles is to keep the list of dependencies that are required to run the model as short as possibile. This choice adds overhead to the process of updating R dependencies, but incurs the benefit of a more maintainable model over the long term.
+
+The process for **updating core model pipeline dependencies** is straightforward: Running `renv::install("<dependency_name>")` and `renv::snapshot()` will ensure that the dependency gets added or updated in `renv.lock`, as long is it is imported somewhere in the model pipeline via a `library(<dependency_name>)` call.
+
+The process for updating **model report dependencies** is more complex, since it requires the use of a separate `reporting` profile:
+
+1. Run `Sys.setenv(RENV_PROFILE = "reporting")` to set the renv profile to `reporting`
+2. Make sure that the dependency is defined in the `DESCRIPTION` file under the `Config/renv/profiles/reporting/dependencies` key
+3. Run `renv::install("<dependency_name>")` to add or update the dependency as necessary
+4. Run `renv::snapshot(type = "explicit")` to update the reporting lockfile with the dependencies defined in the `DESCRIPTION` file
+5. Run `Sys.unsetenv("RENV_PROFILE")` to switch the renv profile back to the default
```

Comment:
I don't _love_ the complexity of this process, but it was the simplest process I could find to manage two lockfiles using R. If we decide to move in this direction, we could consider wrapping these steps in a helper script.